src/utils/network.ts

```
@@ -23,7 +23,7 @@ export const getNetworkId = (network: Network): number => {
 };
 
 export const getNetwork = (networkId: number): Network => {
-  switch (networkId) {
+  switch (Number(networkId)) {
```

Maybe will be better to use toString for more error redundancy but lgtm

---

Sounds good, refactored with https://github.com/yearn/yearn-finance-v3/pull/560/commits/ee17a64d506836ddac93678e0b4187200251ef00

---

I was wondering if we should throw or log the error somewhere, it wasn't the most obvious bug to find. Do you know if we are allowing for network to be "other" and if we do anything with it or would we be better off throwing and logging an error.

```typescript
throw Error(`Unknown Network Id: ${networkId}`);
```

It would have helped with finding this bug

---

Also added `string` to type of `networkId` with https://github.com/yearn/yearn-finance-v3/pull/560/commits/3e8cc2a808d9c6735936eff2a7099d29de99f081

---

I think we can throw an error too yeah sounds smart, and display an alert to change the network to the current selected network in the drop-down. But maybe that alert should be handled in other place and PR.

---

Agreed, refactored with https://github.com/yearn/yearn-finance-v3/pull/560/commits/99a3ceb760022ca043fc19c9ae76c5d19594dce9 (added a few tests as well)